apps/sms/views/conversation/js/compose.js

```diff
@@ -656,9 +656,12 @@ var Compose = (function() {
           }
         });
 
-      //'pick cancelled' is the error thrown when the pick activity app is
-      // canceled normally
-      } else if (errId !== 'ActivityCanceled' && errId !== 'pick cancelled') {
+      // Some possible 'activity cancel' error name:
+      // 'pick cancelled': error thrown when the pick activity canceled in most
+      //   of the apps.
+      // 'cancelled': error thrown when the pick activity canceled in contacts.
+      // 'ActivityCanceled': error thrown when pick activity canceled from menu.
+      } else if (!/cancel/i.test(errId)) {
```

Comment:
Yeah, I remember we were discussing in some bug - ```activity.cancel```, maybe we also need ```activityRequest.cancel``` in addition to ```activityRequest.postError```. For now and for us it's fine like you did I think.